odl/discr/lp_discr.py

```diff
@@ -446,8 +528,16 @@ def __setitem__(self, indices, values):
             shape is allowed as ``values``.
         """
         if values in self.space:
-            self.ntuple.__setitem__(indices, values.ntuple)
+            # For RawDiscretizationVector of the same type, use ntuple directly
+            return self.ntuple.__setitem__(indices, values.ntuple)
+        elif (isinstance(self.space.dspace, Ntuples) and
+              equiv_views(self.asarray(), values) and
```

Comment:
Ahh and I'm slow. We do a `Ntuples` check first. :/

Still I dislike these very specfic optimizations, is there no way to simply push this to the dspace?